spec/controllers/reviews_controller_spec.rb

```diff
@@ -254,49 +254,94 @@ def valid_session
+        end
       end
 
-      it "redirects to the review" do
-        review = Review.create! valid_attributes
-        put :update, {:id => review.to_param, :review => valid_attributes}, valid_session
-        response.should redirect_to(review)
+      describe "with invalid params" do
+        it "assigns the review as @review" do
+          review = Review.create! valid_attributes
+          # Trigger the behavior that occurs when invalid params are submitted
+          Review.any_instance.stub(:save).and_return(false)
+          put :update, {:id => review.to_param, :review => {}}, valid_session
+          assigns(:review).should eq(review)
+        end
+
+        it "re-renders the 'edit' template" do
+          review = Review.create! valid_attributes
+          # Trigger the behavior that occurs when invalid params are submitted
+          Review.any_instance.stub(:save).and_return(false)
+          put :update, {:id => review.to_param, :review => {}}, valid_session
+          response.should render_template("edit")
+        end
       end
     end
 
-    describe "with invalid params" do
-      it "assigns the review as @review" do
-        review = Review.create! valid_attributes
-        # Trigger the behavior that occurs when invalid params are submitted
-        Review.any_instance.stub(:save).and_return(false)
-        put :update, {:id => review.to_param, :review => {}}, valid_session
-        assigns(:review).should eq(review)
+    context 'as a user' do
+      let(:review_user) { create(:user) }
+      let(:ac) { create(:associate_consultant, user: review_user) }
+      before(:each) do
+        @review = Review.create! valid_attributes
+        @review.update_attribute(:associate_consultant_id, ac.id)
+        @review.update_attribute(:review_date, @review.review_date + 1.month)
+        @review.update_attribute(:reviewee, ac.user)
+        set_current_user ac.user
       end
 
-      it "re-renders the 'edit' template" do
-        review = Review.create! valid_attributes
-        # Trigger the behavior that occurs when invalid params are submitted
-        Review.any_instance.stub(:save).and_return(false)
-        put :update, {:id => review.to_param, :review => {}}, valid_session
-        response.should render_template("edit")
+      describe "with valid params" do
+        it "updates the requested review" do
+          Review.any_instance.should_receive(:update_attributes).with({'these' => 'params'})
+          put :update, {:id => @review.to_param, :review => {'these' => 'params'}}, valid_session
+        end
+      end
+
+      it "should notify invitees when the deadline changes" do
+        @review.invitations << FactoryGirl.create_list(:invitation, 5)
+        UserMailer.should_receive(:review_update).exactly(@review.invitations.count).times.and_return(double(deliver: true))
+        put :update, {:id => @review.to_param, :review => {'feedback_deadline' => @review.feedback_deadline - 2.weeks,
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.